US 2: Email Registration Form

MatiasDG539 opened the pull request:
I have completed all the tasks described in the second user story.

###Checklist (Available on Notion, too)
- [x]  Include email input field with validation
- [x]  Display "Next" button (disabled until valid input)
- [x]  Show "Use email instead" / "Use phone instead" toggle
- [x]  Implement real-time input validation
- [x]  Require password and password confirmation
- [x]  Require personal information: phone number, name, username (unique), birthdate, user picture

##Extra:

I have reduce loading time for splash screen on both platforms to 1.5 seconds.

Proof of implementation

Below are the demonstration videos:

    - Web view
    - iOS view (simulated with Xcode)
    
Web view:

https://github.com/user-attachments/assets/0f2ac683-9fc2-41c5-8bb3-2bcf500db83f

iOS view:

https://github.com/user-attachments/assets/8591203e-3a4f-48ee-94da-991f94b26c25

MatiasDG539 committed:
feat: reduce splash screen loading time

MatiasDG539 committed:
feat: implement sign_up page for web and mobile platforms.

MatiasDG539 committed:
feat: reduce splash screen loading time (web)

MatiasDG539 committed:
feat: implement home screen placeholder and remove old index file

MatiasDG539 committed:
style: rename files to kebab-case and adjust spacing for readability (web)

MatiasDG539 committed:
style: rename files to kebab-case

MatiasDG539 committed:
feat: extract splash screen into a component for readability and maintainability

MatiasDG539 committed:
style: format tailwind.config.ts for improved readability

MatiasDG539 committed:
refactor: remove splash screen styles and integrate animations for improved structure

MatiasDG539 committed:
feat: upgrade tailwind css and improve code formatting

MatiasDG539 committed:
Revert "feat: upgrade tailwind css and improve code formatting"

This reverts commit d28fa4b2ab49226e534b8e7ab145b0582ed36aef.

MatiasDG539 committed:
fix(web): incorrectly applied Tailwind v4 update

MatiasDG539 committed:
refactor(web): rename utility classes for Tailwind v4

agustin-sanc reviewed (commented):
Mati! Te voy dejando algunos comments

MatiasDG539 committed:
refactor: rename file to follow naming conventions & add blank spaces

MatiasDG539 committed:
refactor: rename LandingPage to HomeScreen for consistency

MatiasDG539 committed:
refactor: convert SignUpScreen to arrow function syntax

agustin-sanc reviewed (commented):
Mati! Gracias por resolver los comentarios anteriores, te dejo algunos comentarios nuevos.

MatiasDG539 committed:
refactor: import useEffect for improved state management in SignUpPage

MatiasDG539 committed:
feat: add shared forms package with Zod schemas and utilities

MatiasDG539 committed:
refactor: remove unused validation functions and image preview utilities from schemas and utils

MatiasDG539 committed:
fix: update zod dependency to version 4.1.3 in package.json and pnpm-lock.yaml

MatiasDG539 committed:
feat(web): replace manual forms with shared form package

MatiasDG539 committed:
feat(mobile): replace manual forms with shared form package

MatiasDG539 committed:
fix: change pickImage function to synchronous

MatiasDG539 committed:
fix: remove unnecessary blank line in HomeScreen component

MatiasDG539 commented on apps/web/app/auth/sign-up/page.tsx:
```diff
@@ -0,0 +1,8 @@
+import type { FC } from "react";
+
+const SignUp: FC = () => {
+  return <SignUpPage />;
```
> Este return podría ser implícito. De hecho, el componente `SignUpPage` puede ser innecesario, y tener ese código directamente aquí. Por qué lo tenemos en un componente aparte?

Decidí dejarlo como componente y luego importarlo para mejorar la organización y legibilidad del código. Esto permite que el archivo principal sea más claro y facilita el mantenimiento o posibles reutilizaciones futuras.

agustin-sanc replied:
Como acordamos en la call, estaría bueno no hacer esta separación a menos que sea muy necesario en algún caso particular

MatiasDG539 commented:
Hola Agus! Correcciones realiazadas!

>Podrías refactorizar los forms para utilizar React Hook Forms y las validaciones con schemas de Zod?

Con respecto a esto hice lo siguiente: Cree el package `/form` dentro de `/packages` para poder utilizarlos tanto en `/mobile` como en `/web`. Me parecio el enfoque mas prolijo para hacer asi evito tener codigo duplicado.

MatiasDG539 committed:
feat(forms): implement platform-specific logic for web and mobile

- Removed shared forms from packages/
- Added dedicated form logic under apps/web and apps/mobile

MatiasDG539 committed:
feat: add alias for forms in Metro and Next.js configurations on mobile and web platforms

MatiasDG539 committed:
refactor: simplify form handling by removing redundant state management on mobile and web platforms

agustin-sanc commented on apps/mobile/forms/utils.ts:
Las funciones de este archivo podrían ser arrow functions con return implícito para que quede más limpio

agustin-sanc replied:
Estamos usando estos forms o quedaron obsoletos?

agustin-sanc commented on apps/mobile/forms/schemas.ts:
La carpeta `forms` podría estar dentro de la carpeta `components` sin problemas, por las dudas. Generalmente siempre metemos `forms` dentro de `components`

agustin-sanc commented on apps/web/forms/utils.ts:
Estamos usando estas funciones o ya no?

agustin-sanc reviewed (approved):
Mucho mejor Mati, va queriendo. Dejé algunos comments más, dejo aprobado para que mergees la PR después de corregirlos

MatiasDG539 committed:
refactor: rename SignUpPage to SignUpScreen for consistency and clarity

MatiasDG539 committed:
refactor: remove unused form utility functions from mobile and web platforms

MatiasDG539 committed:
refactor: consolidate SignUpPage logic into SignUp component and remove redundant file

MatiasDG539 committed:
refactor(web, mobile): move form schemas into components folder
